ledger/ledger-on-memory/src/main/scala/com/daml/ledger/on/memory/InMemoryLedgerReaderWriter.scala

```diff
@@ -68,9 +83,12 @@ object InMemoryLedgerReaderWriter {
   final class SingleParticipantOwner(
       ledgerId: LedgerId,
       batchingLedgerWriterConfig: BatchingLedgerWriterConfig,
+      preExecute: Boolean,
       participantId: ParticipantId,
       timeProvider: TimeProvider = DefaultTimeProvider,
       stateValueCache: Cache[DamlStateKey, DamlStateValue] = Cache.none,
+      stateValueCacheForPreExecution: Cache[DamlStateKey, (DamlStateValue, Fingerprint)] =
```

Comment:
Removed.

Reply:
It's still here. Just use the below `*Owner` classes instead of this one, please. The only addition you need is creating the dispatcher and an empty state.